datafusion/src/physical_plan/math_expressions.rs

```diff
@@ -116,3 +116,12 @@ math_unary_function!("exp", exp);
 math_unary_function!("ln", ln);
 math_unary_function!("log2", log2);
 math_unary_function!("log10", log10);
+
+/// random SQL function
+pub fn random(_: &[ColumnarValue]) -> Result<ColumnarValue> {
```

Comment:
Would it be more efficient if this is implemented as an `arrow::compute` kernel, so that instead of creating individual scalar values, you could fill a buffer of `n` elements with random data at once?